sqldelight-gradle/android-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/SqlDelightAndroidPlugin.kt

```diff
@@ -76,16 +71,24 @@ class SqlDelightAndroidPlugin : Plugin<Project> {
 
     project.afterEvaluate {
       val ideaDir = File(project.rootDir, ".idea")
-      if (!ideaDir.exists()) return@afterEvaluate
-      val propsDir = File(ideaDir, "sqldelight/${project.projectDir.toRelativeString(project.rootDir)}")
-      propsDir.mkdirs()
+      if (ideaDir.exists()) {
+        val propsDir =
+          File(ideaDir, "sqldelight/${project.projectDir.toRelativeString(project.rootDir)}")
+        propsDir.mkdirs()
+
+        val properties = SqlDelightPropertiesFile(
+            packageName = packageName!!,
+            sourceSets = sourceSets,
+            outputDirectory = buildDirectory.toRelativeString(project.projectDir)
+        )
+        properties.toFile(File(propsDir, SqlDelightPropertiesFile.NAME))
+      }
 
-      val properties = SqlDelightPropertiesFile(
-          packageName = packageName!!,
-          sourceSets = sourceSets,
-          outputDirectory = buildDirectory.toRelativeString(project.projectDir)
+      addMigrationTasks(
+          project = project,
+          sourceSet = sourceSets.flatMap { it.map { relative -> File(project.projectDir, relative) } }.distinct(),
```

Comment:
nit: I think you can hoist the `map` outside of the `flatMap` to reduce some of the cognitive load of this.

Also what is `sourceSets` here? Does replacing `flatMap` with a `flatten()` call work?

Reply:
agreed.

sourceSets will look like 
```
[
  [src/main/sqldelight, src/debug/sqldelight],
  [src/main/sqldelight, src/release/sqldelight]
]
```

so i dont want src/main/sqldelight to appear twice.

Reply:
Ok. So I think `sourceSets.flatten().map { File(project.projectDir, it) }.distinct()` should work then.

Reply:
Or `sourceSets.flatten().distinct().map { File(..) }`